app/controllers/api/v2/anime_controller.rb

```diff
@@ -2,6 +2,21 @@ module Api::V2
   class AnimeController < ApiController
     caches_action :show, expires_in: 1.hour
 
+    def index
+      anime = Anime.preload(:genres)
+
+      if params.key? :filter
+        search = AnimeSearch.new(params[:filter])
+        if search.valid?
+          anime = search.apply_filters(anime)
+        else
+          return render json: search.errors.full_messages
+        end
+      end
+
+      render json: anime.page(params[:page]).per(20)
```

Comment:
I'll fix it by Monday (I'll also check error messages, http standards and
other json api items).

I can also update the docs, I'll just need some proofread for my poor
English :P
Em 07/06/2015 01:14, "Peter Lejeck" <notifications@github.com> escreveu:

> In app/controllers/api/v2/anime_controller.rb
> <https://github.com/hummingbird-me/hummingbird/pull/537#discussion_r31873761>
> :
>
> > @@ -2,6 +2,21 @@ module Api::V2
> >    class AnimeController < ApiController
> >      caches_action :show, expires_in: 1.hour
> >
> > +    def index
> > +      anime = Anime.preload(:genres)
> > +
> > +      if params.key? :filter
> > +        search = AnimeSearch.new(params[:filter])
> > +        if search.valid?
> > +          anime = search.apply_filters(anime)
> > +        else
> > +          return render json: search.errors.full_messages
> > +        end
> > +      end
> > +
> > +      render json: anime.page(params[:page]).per(20)
>
> Just found a [image: :beetle:] on this that needs fixing before anyone
> starts using the endpoint: this needs to be
> AnimeSerializer.new(anime).as_json (I don't believe Api::V2 is required)
> like below, and we will want to include meta for pagination
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hummingbird-me/hummingbird/pull/537/files#r31873761>.
>